src/lexer_tok.c

```diff
@@ -107,15 +107,16 @@ t_error	ft_partition(t_src *src, t_buf *tmp)
 			ft_add_quoted_str(c, src, tmp);
 		else if ((c == ' ' || c == '\t') && tmp->cur_pos > 0)
 			break ;
-		else if ((c == '\n' || c == '|') && tmp->cur_pos > 0)
+		else if ((c == '\n' || c == '|' || c == '<' || c == '>')
+			&& tmp->cur_pos > 0)
 		{
 			ft_unget_char(src);
 			break ;
 		}
 		else
 		{
 			ft_add_to_buf(c, tmp);
```

Comment:
what i could do is write another function who - as you proposed - receives t_src instead of ft_add_to_buf(). In case the next char is the same redirect (use ft_peek_char), then add it as well to buf, if not just add the current char.

Reply:
Added ft_add_double_redirect(t_src *src, char *c, t_buf *tmp) to create one token out of << or >> respectively.